File: app/src/main/java/oak/shef/ac/uk/testrunningservicesbackgroundrelaunched/SensorRestarterBroadcastReceiver.java
package oak.shef.ac.uk.testrunningservicesbackgroundrelaunched;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

/**
 * Created by fabio on 24/01/2016.
 */
public class SensorRestarterBroadcastReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(SensorRestarterBroadcastReceiver.class.getSimpleName(), "Service Stops! Oooooooooooooppppssssss!!!!");
        int val = intent.getIntExtra(MainActivity.CURRENT_PROGRESS, 0);
        context.startService(new Intent(context, SensorService.class).putExtra(MainActivity.CURRENT_PROGRESS, val));
    }

}

File: app/src/main/java/oak/shef/ac/uk/testrunningservicesbackgroundrelaunched/SensorRestarterBroadcastReceiver.kt
package oak.shef.ac.uk.testrunningservicesbackgroundrelaunched

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * Created by fabio on 24/01/2016.
 */
class SensorRestarterBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Log.i(SensorRestarterBroadcastReceiver::class.java.simpleName, "Service Stops! Oooooooooooooppppssssss!!!!")
        context.startService(Intent(context, SensorService::class.java))
    }

}
